docs/kb/integration/defect-dojo-integration.md

```diff
@@ -0,0 +1,116 @@
+        'scan_type': (None, 'Semgrep JSON Report'),
+        'product_name': (None, product_name),
+        'engagement_name': (None, engagement_name),
+    }
+
+    endpoint = '/api/v2/import-scan/' if is_new_import else '/api/v2/reimport-scan/'
+    r = requests.post(
+        url + endpoint,
+        files=multipart_form_data,
+        headers={
+            'Authorization': 'Token ' + token,
+        }
+    )
+    if r.status_code != 200:
+        sys.exit(f'Post failed: {r.text}')
+    print(r.text)
+```
+
+The full version of this Python script can be found in this [repository](https://github.com/r2c-CSE/semgrep-utilities/blob/main/integrations/defectdojo/import_semgrep_to_defect_dojo.py).
+It is ready to execute.
+
+### Running the script
+
+To run the script:
+```
+python3 integrations/defectdojo/import_semgrep_to_defect_dojo.py --host DOJO_URL --product PRODUCT_NAME --engagement ENGAGEMENT_NAME --report REPORT_FILE 
+```
+Where:
+* `DOJO_URL` is the URL where DefectDojo is.
+* `PRODUCT_NAME` is the DefectDojo product name.
+* `ENGAGEMENT_NAME` is the DefectDojo engagement name for that product.
+* `REPORT_FILE` is the Semgrep report path
+
+## Integrate Semgrep and DefectDojo in a pipeline
+
+From a DevSecOps point of view is crucial to integrate security scans and dump results to DefectDojo in the same pipeline.
+An example of GitLab job importing results to DefectDojo taking as input a Semgrep report looks like this:
+
+```
+import-semgrep-to-defectdojo:
+  stage: import
+  image: python:3.9-bullseye
+  script:
+    - echo "Importing Semgrep scan to DefectDojo"
+    - pip3 install requests
+    - curl -O https://raw.githubusercontent.com/r2c-CSE/semgrep-utilities/main/integrations/defectdojo/import_semgrep_to_defect_dojo.py
+    - python3 import_semgrep_to_defect_dojo.py --host $DEFECTDOJO_URL --product $PRODUCT --engagement semgrep --report report.json || true
+  rules:
+    # Scan changed files in MRs, (diff-aware scanning):
+    - if: $CI_MERGE_REQUEST_IID
+    # Scan mainline (default) branches and report all findings.
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+  tags:
+    - defectdojo
+  variables:
+    DEFECT_DOJO_API_TOKEN: $DEFECT_DOJO_API_TOKEN
+```
+
+### Adding checksum validation
+To verify the job downloads the correct Python script, it is essential, from a security point of view, to check the integrity with a checksum such as SHA 256 code. It avoids the execution of tampered 
+scripts.
+
+```
+    - echo $IMPORT_SEMGREP_TO_DEFECTDOJO_SHA_CHECKSUM > sha-import-dd.tmp
+    - shasum -a 256 -U -c sha-import-dd.tmp
+```
+Where:
+* `IMPORT_SEMGREP_TO_DEFECTDOJO_SHA_CHECKSUM` is the result of executing `shasum -a 256 -U import_semgrep_to_defect_dojo.py`
+
+## Conclusions
+Dump Semgrep security results to [DefectDojo](https://www.defectdojo.com/) can speed up vulnerability management operations in an organization. Integrating scans and imports in the same CI pipeline should 
```

Comment:
```suggestion
Dumping Semgrep scan results to [DefectDojo](https://www.defectdojo.com/) can speed up vulnerability management operations in an organization. Integrating scans and imports in the same CI pipeline should be a high priority for DevSecOps programs.
```

* I think there was another unintentional newline, so I just copied the line below.
* Are you saying that "integrating scans and imports in the same CI pipeline" prevents the data from being tampered?